Use a test profile, and fix some tests

kinow opened the pull request:
Right now I have one test failing. This PR includes a `nodeps` profile as discussed in #130, for running the unit tests that do not require external accounts in AWS/Azure/etc, nor certain dependencies such as Kakadu or Selenium.

There are also several tests that use font family, and not family name. The problem with that is that the value changes quite a bit in Linux/Windows. So this PR also addresses this problem, by using the font name.

The Helvetica font is not installed in Ubuntu, or at least is not easily installed. Tried the `ms-mscorefonts-installer` but didn't get the Helvetica font, so changed by Arial.

adolski commented on pom.xml:
```diff
@@ -526,4 +526,33 @@
+                <exclude>Amazon*Test</exclude>
+                <exclude>Kakadu*Test</exclude>
+                <exclude>Redis*Test</exclude>
+                <exclude>ControlPanelTest</exclude>
```
I think the complete list would be:

    <exclude>AmazonS3*Test</exclude>
    <exclude>AzureStorage*Test</exclude>
    <exclude>FfmpegProcessorTest</exclude>
    <exclude>GraphicsMagickProcessorTest</exclude>
    <exclude>ImageMagickProcessorTest</exclude>
    <exclude>KakaduProcessorTest</exclude>
    <exclude>OpenJpegProcessorTest</exclude>
    <exclude>RedisCacheTest</exclude>

I excluded ControlPanelTest as I switched it to use HtmlUnit instead of Firefox+geckodriver, so it no longer has external dependencies (although it's still not working and marked as `@Ignore`).

kinow replied:
Oh, so looking at the complete list, I think we can actually keep Ffmpeg, GraphicsMagick, ImageMagick, and OpenJpeg? Or maybe have another profile to include these...

Looks like we have 1) minimum of tests that can be executed in a blank box with Java + Maven, 2) a few more tests that require installation of open source / free tools & libraries (e.g. ffmpeg, graphicsmagick, imagemagick, openjpeg), 3) other tests that require licences or paid tools (e.g. kakadu, amazon, azure).

For the Travis CI tests, we can include 1) + 2). For 2) we'd need just to use a Ubuntu node, and [install some packages](https://docs.travis-ci.com/user/installing-dependencies/).

So do you think we should keep all tests executing by default, have a `nodeps` profile for developers and users to try (with your list above), and another profile for continuous integration (maybe `ci` ?) ?

adolski replied:
Sorry, I was thinking of everything with external dependencies, but yeah, that's the complete list of everything with non-free dependencies.

> So do you think we should keep all tests executing by default, have a nodeps profile for developers and users to try (with your list above), and another profile for continuous integration (maybe ci ?) ?

How about: `freedeps` (used with CI), and then either `nodeps` or `alldeps`?

I'm neutral -- whatever you want to go with.

kinow replied:
Done! Using `freedeps` and `nodeps`.

kinow commented:
Hi @adolski

>Do you want to update the testing section of the readme as well? No obligation, I'm happy to do it.

Oh, good idea! I can send a PR in a few hours with some text for review.

>I think the complete list would be:

Will try with these values when I arrive at work, and update this PR. Thanks!!!

kinow commented:
Rebase'd commit against develop branch. Added `freedeps` profile. Ordered alphabetically the `nodeps` dependencies. Updated README with instructions for running tests.

Right now, here's what I get in my environment:

`mvn clean test -Pfreedeps`

Failed tests: 
  StandaloneEntryTest.testMainWithPurgeExpiredCacheArg:410 expected:<1> but was:<0>
  Java2DUtilTest.testApplyOverlayWithStringOverlay:194

Tests run: 1294, Failures: 2, Errors: 0, Skipped: 21

`mvn clean test -Pnodeps`

Failed tests: 
  StandaloneEntryTest.testMainWithPurgeExpiredCacheArg:410 expected:<1> but was:<0>
  Java2DUtilTest.testApplyOverlayWithStringOverlay:194

Tests run: 1222, Failures: 2, Errors: 0, Skipped: 20
